common/src/main/java/org/embeddedt/modernfix/common/mixin/perf/dynamic_resources/SoundBufferLibraryMixin.java

```diff
@@ -0,0 +1,47 @@
+package org.embeddedt.modernfix.common.mixin.perf.dynamic_resources;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.RemovalNotification;
+import com.mojang.blaze3d.audio.SoundBuffer;
+import net.minecraft.client.sounds.SoundBufferLibrary;
+import net.minecraft.resources.ResourceLocation;
+import org.embeddedt.modernfix.annotation.ClientOnlyMixin;
+import org.embeddedt.modernfix.dynamicresources.DynamicSoundHelpers;
+import org.embeddedt.modernfix.ModernFix;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Mutable;
+import org.spongepowered.asm.mixin.Shadow;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.Map;
+
+@Mixin(SoundBufferLibrary.class)
+@ClientOnlyMixin
+public abstract class SoundBufferLibraryMixin {
+
+    private static final boolean debugDynamicSoundLoading = Boolean.getBoolean("modernfix.debugDynamicSoundLoading");
+
+    @Shadow @Final @Mutable
+    private Map<ResourceLocation, CompletableFuture<SoundBuffer>> cache =
+    CacheBuilder.newBuilder()
+    .maximumSize(DynamicSoundHelpers.MAX_SOUND_COUNT)
+    .expireAfterAccess(DynamicSoundHelpers.MAX_SOUND_LIFETIME_SECS, TimeUnit.SECONDS)
+    // Excessive use of type hinting due to it assuming Object as the broadest correct type
+    .<ResourceLocation, CompletableFuture<SoundBuffer>>removalListener(this::onSoundRemoval)
+    .<ResourceLocation, CompletableFuture<SoundBuffer>>build()
+    .asMap();
+
+    private <K extends ResourceLocation, V extends CompletableFuture<SoundBuffer>> void onSoundRemoval(RemovalNotification<K, V> notification) {
+        if(!debugDynamicSoundLoading) {
+            notification.getValue().thenAccept(SoundBuffer::discardAlBuffer);
+            return;
+        }
+        notification.getValue().thenAccept(SoundBuffer::discardAlBuffer);
+        K k = notification.getKey();
+        if(k == null)
+            return;
+        ModernFix.LOGGER.warn("Evicted sound {}", k);
```

Comment:
Any reason why this isn't tied to `debugDynamicSoundLoading` being enabled?

Reply:
It early returns but I suppose I can wrap it into an else.

Reply:
```suggestion
        if(debugDynamicSoundLoading)
            ModernFix.LOGGER.warn("Evicted sound {}", k);
```

Reply:
*facepalm* oops.

Reply:
Wait, no I can just wrap everything into the if except the sound close method.

Reply:
Force-pushed.